src/hello-visualization/app.js

```diff
@@ -170,51 +169,52 @@ angular.module('app', []).component('app', {
       halyard.addTable(tableMovie);
 
       // Add web data
-      $http.get('https://gist.githubusercontent.com/carlioth/b86ede12e75b5756c9f34c0d65a22bb3/raw/e733b74c7c1c5494669b36893a31de5427b7b4fc/MovieInfo.csv')
-        .then((data) => {
-          const table = new Halyard.Table(data.data, { name: 'MoviesInfo', delimiter: ';', characterSet: 'utf8' });
-          halyard.addTable(table);
-        })
-        .then(() => {
-          enigma.create(config).open().then((qix) => {
-            this.connected = true;
-            this.connecting = false;
-            qix.createSessionAppUsingHalyard(halyard).then((result) => {
-              app = result;
-              result.getAppLayout()
-                .then(() => {
-                  result.createSessionObject(scatterplotProperties).then((model) => {
-                    scatterplotObject = model;
+      (async () => {
+        const data = await $http.get('https://gist.githubusercontent.com/carlioth/b86ede12e75b5756c9f34c0d65a22bb3/raw/e733b74c7c1c5494669b36893a31de5427b7b4fc/MovieInfo.csv');
+        const table = new Halyard.Table(data.data, { name: 'MoviesInfo', delimiter: ';', characterSet: 'utf8' });
+        halyard.addTable(table);
+        let qix;
+        try {
+          qix = await enigma.create(config).open(); // lägg till try t
+          this.connected = true;
+          this.connecting = false;
+        } catch (error) {
+          this.error = 'Could not connect to QIX Engine';
+          this.connecting = false;
+        }
+        // läg till try cath
 
-                    const update = () => scatterplotObject.getLayout().then((layout) => {
-                      paintScatterPlot(layout);
-                    });
+        let result;
```

Comment:
You can skip `result` and use app directly. So, 
`app = await qix.createSessionAppUsingHalyard(halyard);`